src/Reflection/ReflectionClass.php

```diff
@@ -251,19 +262,54 @@ function (ClassMethod $methodNode) : ReflectionMethod {
                 $this->node->getMethods()
             ),
             ...array_map(
-                function (ReflectionClass $trait) : array {
-                    return array_map(function (ReflectionMethod $method) use ($trait) : ReflectionMethod {
-                        $methodAst = $method->getAst();
-                        assert($methodAst instanceof ClassMethod);
-
-                        return ReflectionMethod::createFromNode(
-                            $this->reflector,
-                            $methodAst,
-                            $this->declaringNamespace,
-                            $trait,
-                            $this
-                        );
-                    }, $trait->getMethods());
+                function (ReflectionClass $trait) use ($traitAliases, $traitPrecedences) : array {
+                    return array_merge(
+                        [],
+                        ...array_filter(
+                            array_map(
+                                function (ReflectionMethod $method) use ($traitAliases, $traitPrecedences) : ?array {
+                                    $methodAst = $method->getAst();
+                                    assert($methodAst instanceof ClassMethod);
+
+                                    $aliases = [];
+                                    foreach ($traitAliases as $aliasMethodName => $traitAliasDefinition) {
+                                        if (sprintf('%s::%s', $method->getDeclaringClass()->getName(), $method->getName()) !== $traitAliasDefinition) {
```

Comment:
the `%s::%s` approach for lookups seems very fragile: I see that it is pre-existing, but indeed unfortunate storage mechanism.

I would stash this out into a `private function methodHash(ReflectionClass $declaringClass, ReflectionMethod $method)` or such

Reply:
Seems to be done in a closure below, called `$formatMethodName`: we could use the same in `private` API